components/api/src/main/java/com/hotels/styx/api/messages/FullHttpRequest.java

```diff
@@ -0,0 +1,694 @@
+            return headers.build();
+        }
+
+        public HttpVersion version() {
+            return version;
+        }
+
+        public T body() {
+            return body;
+        }
+
+        public Builder<T> version(HttpVersion version) {
+            this.version = requireNonNull(version);
+            return this;
+        }
+
+        public Builder<T> method(HttpMethod method) {
+            this.method = requireNonNull(method);
+            return this;
+        }
+
+        public List<HttpCookie> cookies() {
+            return ImmutableList.copyOf(cookies);
+        }
+
+        /**
+         * Adds a response cookie (adds a new Set-Cookie header).
+         *
+         * @param cookie cookie to add
+         * @return {@code this}
+         */
+        public Builder<T> addCookie(HttpCookie cookie) {
+            cookies.add(checkNotNull(cookie));
+            return this;
+        }
+
+
+        /**
+         * Adds a response cookie (adds a new Set-Cookie header).
+         *
+         * @param name  cookie name
+         * @param value cookie value
+         * @return {@code this}
+         */
+        public Builder<T> addCookie(String name, String value) {
+            return addCookie(HttpCookie.cookie(name, value));
+        }
+
+        /**
+         * Removes a cookie if present (removes its Set-Cookie header).
+         *
+         * @param name name of the cookie
+         * @return {@code this}
+         */
+        public Builder<T> removeCookie(String name) {
+            cookies.stream()
+                    .filter(cookie -> cookie.name().equalsIgnoreCase(name))
+                    .findFirst()
+                    .ifPresent(cookie -> cookies.remove(cookie));
+
+            return this;
+        }
+
+        public Builder<T> clientAddress(InetSocketAddress clientAddress) {
+            this.clientAddress = requireNonNull(clientAddress);
+            return this;
+        }
+
+        /**
+         * Removes the header with the specified name.
+         *
+         * @param name The name of the header to remove
+         * @return {@code this}
+         */
+        public Builder<T> removeHeader(CharSequence name) {
+            headers.remove(name);
+            return this;
+        }
+
+        public Builder<T> secure(boolean secure) {
+            this.secure = secure;
+            return this;
+        }
+
+        /**
+         * Enable validation of uri and some headers.
+         *
+         * @return {@code this}
+         */
+        public Builder<T> disableValidation() {
+            this.validate = false;
+            return this;
+        }
+
+
+        public Builder<T> enableKeepAlive() {
+            return header(CONNECTION, KEEP_ALIVE);
+        }
+
+        public FullHttpRequest<T> build() {
+            if (validate) {
+                ensureContentLengthIsValid();
+                ensureMethodIsValid();
+            }
+
+            setHostHeader();
```

Comment:
This too will prevent some use cases where we deliberately might *not* want to have the host header present. Such as adapting wire-mock onto Styx server framework. Perhaps move into the *validate* block or something?